### PR TITLE
Add release-checklist issue template

### PR DESCRIPTION
**Pull request by iameskild**
Related to https://github.com/nebari-dev/governance/issues/2

> Please remove anything marked as optional that you don't need to fill in.
> Choose one of the keywords preceding to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666).
> If there is no issue, remove the line. Remove this note after reading.

## Changes introduced in this PR:

- Enables maintainers to easily create a release checklist that can guide them through the release process.

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [ ] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [ ] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  - [Do we need to update the `dask` versions in the `qhub-dask`?](https://github.com/conda-forge/qhub-dask-feedstock/blob/main/recipe/meta.yaml#L13-L16)
+  - Will there be an accompanying blog post?
+- [ ] Prepare for the release.
+  - [ ] Announce build freeze.
+  - [ ] Release Candidate (RC) cycle.
+    - Is this a hotfix?
+      - [ ] Create a new branch off of the last version tag.
+        - Use this branch to cut the pre-release and the "official" release.
+      - [ ] `git cherry-pick` the commits that should be included.
+    - [ ] [Perform end-to-end testing.](https://docs.qhub.dev/en/latest/source/dev_guide/release.html#pre-release-checklist)
```
[the remark this answers is not in the document]
Comment (iameskild) [reply .github/ISSUE_TEMPLATE/release-checklist.md:52]: I agree it's confusing, I'm just a bit concerned about the length and complexity of the one checklist given the matrix of possible "flavors" that need to be tested, especially for an important release. For example, the recent hotfix only relied on CI (and integration testing) as a means of validating the fixes worked, but the recent 0.4.0 release required much more intensive user testing. Perhaps we could add another issue template for a user testing checklist, essentially something like this: https://github.com/Quansight/qhub/issues/976

[the remark this answers is not in the document]
Comment (iameskild) [reply .github/ISSUE_TEMPLATE/release-checklist.md:52]: I'll remove the checklist that lives in the docs and replace it with a link to this checklist.

**Comment by iameskild** [comment]
Hi @trallard I've added the testing checklist issue template and updated the docs to point to reference these checklist templates. Let me know if you have any other feedback :)

**Review by trallard — APPROVED** [review]
LGTM 🚀

**Pull request merged**